src/workloads/execution/FilterWithLogicalExpressions.yml

```diff
@@ -0,0 +1,162 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: >
+  This workload stresses the query execution engine by running queries with complex logical
+  expressions that never match a document in the collection.
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection Collection0
+  DocumentCount: &DocumentCount 1e6
+  MaxPhases: &MaxPhases 5
+
+Actors:
+# Phase 0: Insert documents into the collection.
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document:
+          # NOTE: These documents will not produce a match for the filter below, forcing the query
+          # to scan the entire collection.
+          a: 1
+          b: 1
+          c: 1
+          d: 1
+
+# Phase 1: Ensure all data is synced to disk.
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+
+# Phase 2: Run query with wide expression.
+- Name: RunWideQuery
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 10
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {
+              $expr: {
+                $or: [
+                  {$eq: ["$a", 10]},
```

Comment:
Do we want to test dot path here as well?

Reply:
+1

I don't know if it would go into this Genny workload or not, but I'd love to see two more test cases:

- A test with a deep field path expression, like "$a.b.c.d.e.f.g".
- A test that combines a wide logical expression with deep field paths, e.g. an $or with 10 clauses where each clause has a deep field path expression.

Reply:
I think that adding a wide query with deep fieldpaths is a good idea. I am not sure about the first case, though. 7-depth fieldpath traversal seems too synthetic. We can add a query with a lower number of fields (something like 3), but I think we already have queries like this in the `*Subpath*` microbenchmarks.

Reply:
> 7-depth fieldpath traversal seems too synthetic.

These are all highly synthetic. If we have a performance problem which manifests specifically due to how we construct plans for deep paths, it seems worth at least knowing about. If we had data saying that customers almost never use field paths deeper than 3, then I could see an argument for omitting this. But since we don't actually know how deep field paths tend to go it seems worthwhile to have just one benchmark stressing this dimension.

Actually, it would be two additional benchmarks: One for a deep field path expression in the agg expression language and a corresponding test for the match expression language.

Reply:
Sounds good. I have modified the "wide with deep fieldpaths" to use 7-depth fieldpaths and added a query with a single equality predicate with deep fieldpaths. I suggest to add other cases related to scalability along various axis in future pull requests.